training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -57,7 +62,8 @@ public User update(@PathVariable Long id, @RequestBody User user) throws UserNot
             throw  new UserIdMismatchException();
         }
 
-        userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        User actualUser = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        user.setPassword(actualUser.getPassword());
```

Comment:
You're not encrypting the password here. I would add the encrypt password to the setter so you don't have to remember doing it everywhere.

Reply:
I made it that way because the card said that the password should not be updated in the same resource as a user. So now when somebody updates a user, it doesn't matter what they do with the password (if they change it or not), it will remain as the original one. To change the password it's necesaty to call _/api/users/editPass/{userId}_. Should I change this?
>Siempre debe modificarse la password mediante un recurso distinto al de actualización de un User.

Reply:
That's right but what you need to do is set each field manually, skipping the password one :)

Reply:
Great!